receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -13,82 +13,182 @@ import (
 
 	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
 	cExtractor "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/extractors"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
 	"go.uber.org/zap"
 	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
-type kubeletSummaryProvider struct {
-	logger   *zap.Logger
-	hostIP   string
-	hostPort string
-	client   *kubeletutil.KubeletClient
-	hostInfo host.Info
+// KubeletProvider Represents interface to kubelet.
+type KubeletProvider interface {
+	GetClient() (*kubeletutil.KubeletClient, error)
+	GetSummary() (*stats.Summary, error)
+}
+
+type kubeletProvider struct {
+	logger *zap.Logger
+	hostIP string
+	port   string
```

Comment:
Change the field name `port` to `hostPort` to maintain the code same as kubeletSummaryProvider

Reply:
Addressed.